starknet_py/cairo/serialization/data_serializers/uint256_serializer.py

```diff
@@ -39,14 +45,32 @@ def deserialize_with_context(self, context: DeserializationContext) -> int:
         return (high << 128) + low
 
     def serialize_with_context(
-        self, context: SerializationContext, value: int
+        self, context: SerializationContext, value: Union[int, Uint256Dict]
     ) -> Generator[int, None, None]:
+        context.ensure_valid_type(value, isinstance(value, (int, dict)), "int or dict")
+        if isinstance(value, int):
+            yield from self._serialize_from_int(value)
+        else:
+            yield from self._serialize_from_dict(context, value)
+
+    @staticmethod
+    def _serialize_from_int(value: int) -> Generator[int, None, None]:
         uint256_range_check(value)
         result = (value % 2**128, value // 2**128)
         yield from result
 
+    def _serialize_from_dict(
+        self, context: SerializationContext, value: Uint256Dict
+    ) -> Generator[int, None, None]:
+        with context.push_entity("low"):
+            self._ensure_valid_uint128(value["low"], context)
+            yield value["low"]
+        with context.push_entity("high"):
+            self._ensure_valid_uint128(value["high"], context)
+            yield value["high"]
+
     @staticmethod
-    def _ensure_valid_uint128(value: int, context: DeserializationContext):
+    def _ensure_valid_uint128(value: int, context: Context):
         context.ensure_valid_value(
             0 <= value < U128_UPPER_BOUND, "expected value in range [0;2^128)"
```

Comment:
```suggestion
            0 <= value < U128_UPPER_BOUND, "expected value in range [0;2**128)"
```
I think using `**` for powers is more readable for python users.